src/CistromeHGWConsumer.js

```diff
@@ -0,0 +1,186 @@
+import React, { useRef, useEffect, useState, useMemo, useCallback, useContext } from 'react';
+import PubSub from 'pubsub-js';
+import isEqual from 'lodash/isEqual';
+
+import { HiGlassComponent } from 'higlass';
+import higlassRegister from 'higlass-register';
+import StackedBarTrack from 'higlass-multivec/es/StackedBarTrack.js';
+
+import { EVENT } from './utils/constants.js';
+import { InfoContext, ACTION } from './utils/contexts.js';
+import { selectRows } from './utils/select-rows.js';
+import TrackWrapper from './TrackWrapper.js';
+import Tooltip from './Tooltip.js';
+
+import { processWrapperOptions, getTrackWrapperOptions, updateRowSortOptions } from './utils/options.js';
+import { 
+    getHMTrackIdsFromViewConfig, 
+    getSiblingVPHTrackIdsFromViewConfig,
+    updateViewConfigOnSelectGenomicInterval,
+    updateViewConfigOnSelectRowsByTrack,
+    getHMSelectedRowsFromViewConfig
+} from './utils/viewconf.js';
+
+import './CistromeHGW.scss';
+
+higlassRegister({
+    name: 'StackedBarTrack',
+    track: StackedBarTrack,
+    config: StackedBarTrack.config,
+});
+
+const hgOptionsBase = {
+    bounded: true,
+    pixelPreciseMarginPadding: true,
+    containerPaddingX: 0,
+    containerPaddingY: 0,
+    sizeMode: 'default'
+};
+
+/**
+ * CistromeHGW passes its props through, and wraps this component with the context provider.
+ */
+export default function CistromeHGWConsumer(props) {
+
+    const { viewConfig, options: optionsRaw } = props;
+
+    const hgRef = useRef();
+    const drawRef = useRef({});
+
+    const [options, setOptions] = useState({});
+    const [trackIds, setTrackIds] = useState([]);
+    const [siblingTrackIds, setSiblingTrackIds] = useState({});
+
+    const context = useContext(InfoContext);
+
+    /*
+     * Function to call when the view config has changed.
+     * Updates the array of `horizontal-multivec` track IDs.
+     * Updates the mapping of `horizontal-multivec` track IDs 
+     * to sibling `viewport-projection-horizontal` track IDs.
+     */
+    const onViewConfig = useCallback((newViewConfig) => {
+        const newTrackIds = getHMTrackIdsFromViewConfig(newViewConfig);
+        const newSiblingTrackIds = {};
+        for(let trackId of newTrackIds) {
+            // Each trackId is actually an object `{ viewId, trackId }`, which is why we want trackId.trackId.
+            newSiblingTrackIds[trackId.trackId] = getSiblingVPHTrackIdsFromViewConfig(newViewConfig, trackId.trackId);
```

Comment:
Do you think it would be more understandable when we change the `newTrackIds` and `trackId` to more general names (e.g., newTrackInfos, trackInfo) since these variables contain information of each track other than only the trackId?

Reply:
yes definitely

Reply:
Okay, I will update after I merge!